plugin/src/software/aws/toolkits/eclipse/amazonq/chat/ChatCommunicationManager.java

```diff
@@ -435,14 +439,35 @@ private CompletableFuture<Void> handleCancellation(final String tabId) {
 
     private void sendErrorToUi(final String tabId, final Throwable exception) {
         String errorTitle = "An error occurred while processing your request.";
-        String errorMessage = String.format("Details: %s", exception.getMessage());
+        String errorMessage;
+        if (exception instanceof ResponseErrorException) {
+            ResponseError responseError = ((ResponseErrorException) exception).getResponseError();
+            if (responseError.getData() instanceof JsonObject) {
+                JsonObject responseData = (JsonObject) responseError.getData();
+                if (responseData.has("type") && "answer".equals(responseData.get("type").getAsString())
+                    && responseData.has("body")) {
+                    String body = responseData.get("body").getAsString();
+                    errorMessage = String.format("Details: %s", body);
+                } else {
+                    errorMessage = String.format("Details: %s", responseError.getMessage());
+                }
+            } else {
+                errorMessage = String.format("Details: %s", responseError.getMessage());
+            }
+        } else if (exception.getCause() instanceof ResponseErrorException) {
+            ResponseError responseError = ((ResponseErrorException) exception.getCause()).getResponseError();
+            errorMessage = String.format("Details: %s", responseError.getMessage());
```

Comment:
probably need a null check here too for responseError because if response error is null, we will want to default to basic ex.message